packages/x-scheduler-headless/src/utils/SchedulerStore/SchedulerStore.ts

```diff
@@ -63,21 +64,24 @@ export class SchedulerStore<
     instanceName: string,
     mapper: SchedulerParametersToStateMapper<State, Parameters>,
   ) {
+    const timezone = parameters.timezone ?? 'default';
+
     const schedulerInitialState: SchedulerState<TEvent> = {
       ...SchedulerStore.deriveStateFromParameters(parameters, adapter),
       ...buildEventsState(parameters, adapter),
       ...buildResourcesState(parameters),
       preferences: DEFAULT_SCHEDULER_PREFERENCES,
       adapter,
       occurrencePlaceholder: null,
-      nowUpdatedEveryMinute: adapter.now('default'),
+      nowUpdatedEveryMinute: getNowInRenderTimezone(adapter, timezone),
       pendingUpdateRecurringEventParameters: null,
-      timezone: parameters.timezone ?? 'default',
+      timezone,
       visibleResources: new Map(),
       visibleDate:
         parameters.visibleDate ??
-        parameters.defaultVisibleDate ??
-        adapter.startOfDay(adapter.now('default')),
+        (parameters.defaultVisibleDate
+          ? adapter.setTimezone(parameters.defaultVisibleDate, timezone)
```

Comment:
Same as other models, I think we should change the timezone of `parameters.defaultVisibleDate` before storing it in the state.
The selector will take care of it 👍